### ISSUE TITLE
Refactor models.R to use Keras functional API

### ISSUE THREAD

**Issue by davidrsch** [davidrsch/lstm_re#4]
﻿## Summary
`app/logic/models.R` uses the older `keras_model_sequential` API. `lstm_uts` uses the modern Keras functional API (`keras_input` + `keras_model`) which is more explicit, composable, and aligned with current Keras 3 best practices.

## Current approach (`models.R`)
```r
model <- keras_model_sequential() |>
  layer_lstm(units = ..., input_shape = c(...), return_sequences = TRUE) |>
  ...
```

## Target approach (from `lstm_uts/app/logic/model.R`)
```r
input <- keras_input(shape = c(inp_amount, features))
hidden <- input |>
  layer_lstm(units = lstm)
output <- hidden |>
  layer_dense(units = horizon) |>
  layer_reshape(c(horizon, 1))
model <- keras_model(inputs = input, outputs = output)
model |> compile(loss = "mse", optimizer = "adam")
```

## Changes Required
- Replace `keras_model_sequential` with `keras_input` + `keras_model` in `app/logic/models.R`
- Remove `layer_reshape` workaround from current sequential approach
- Ensure multi-layer support still works cleanly under functional API
- Update imports in `box::use()` accordingly
- Move model compilation (`compile(loss = "mse", optimizer = "adam")`) into `create_model` (currently done elsewhere)

## References
- `lstm_uts/app/logic/model.R`  `get_model()` function
